### PR TITLE
multi: sidecar bug fixes and additional tests

### PR DESCRIPTION
**Pull request by Roasbeef**
Fixes some basic bugs, marking WIP until the set of unit tests is up.

### CODE REVIEW

File: auto_sidecar.go [hunk 1]
```diff
@@ -1,21 +1,45 @@
...
 	"github.com/lightninglabs/pool/order"
 	"github.com/lightninglabs/pool/sidecar"
+	"github.com/lightningnetwork/lnd/keychain"
 	"github.com/lightningnetwork/lnd/lnwire"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
 )
 
+// MailBox is an interface that abstracts over the HashMail functionality to
+// represents a generic mailbox that both sides will use to communicate with
```
Comment (guggero) [inline auto_sidecar.go:19]: nit: s/represents/represent/

**Review by guggero — COMMENTED** [review]
Nice refactor! Did a quick pass and looks great so far :tada:

**Review by halseth — APPROVED** [review]
LGTM, great changes and additional tests 👍

File: auto_sidecar.go [hunk 2]
```diff
@@ -84,79 +107,109 @@ func deriveStreamID(ticket *sidecar.Ticket, provider bool) ([64]byte, error) {
...
 }
 
-// recvSidecarPkt attempts to receive a new sidecar packet from the opposite
-// party using their registered cipherbox stream.
-func (a *SidecarAcceptor) recvSidecarPkt(ticket *sidecar.Ticket,
-	provider bool) (*sidecar.Ticket, error) {
+// SidecarNegotiator is a sub-system that uses a mailbox abstraction between a
+// provider and recipient of a sidecar channel to complete the manual steps in
+// automated manner.
+type SidecarNegotiator struct {
```
[the remark this answers is not in the document]
Comment (Roasbeef) [reply auto_sidecar.go:175]: It's already in its own file, but before this change all the methods were a part of the existing sidecar acceptor code

**Review by guggero — APPROVED** [review]
Great fixes and tests, LGTM :100:

**Pull request merged**